packages/edit-navigation/src/components/menu-editor/block-editor-panel.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * WordPress dependencies
+ */
+import { useDispatch } from '@wordpress/data';
+import { useEffect } from '@wordpress/element';
+
+import {
+	BlockList,
+	BlockToolbar,
+	NavigableToolbar,
+	ObserveTyping,
+	WritingFlow,
+} from '@wordpress/block-editor';
+import { __ } from '@wordpress/i18n';
+import { Button, Panel, PanelBody } from '@wordpress/components';
+
+export default function BlockEditorPanel( { menuId, saveBlocks } ) {
+	const { clearSelectedBlock } = useDispatch( 'core/block-editor' );
+
+	// Clear the selected block when the menu is changed.
+	// Block selection isn't cleared implicity by the block-editor store.
+	// Dispatching this action fixes an issue where the toolbar
+	// for a block continues to be displayed after it no longer exists.
+	useEffect( () => {
+		clearSelectedBlock();
+	}, [ menuId ] );
+
+	return (
+		<Panel
+			header={
+				<Button isPrimary onClick={ saveBlocks }>
+					{ __( 'Save navigation' ) }
+				</Button>
+			}
+		>
+			<PanelBody title={ __( 'Navigation menu' ) }>
+				<NavigableToolbar
```

Comment:
Wouldn't it be fine to just render the toolbar slot in a fixed position?

Reply:
Using the `BlockToolbar` component mirrors the way the fixed toolbar is implemented in the post editor. My thoughts were that this is preferable as the movers are implemented differently in `BlockToolbar` component—they're inline buttons. Whereas with the Slot, the movers reveal and have a drag handle, which doesn't seem like the right behaviour for an inline toolbar.